### PR TITLE
refactor resources

### PR DESCRIPTION
**Pull request by whit537**
Follow-on from #481. Rebase on master once that lands.

### CODE REVIEW

**Comment by whit537** [comment]
Bah. I give up for tonight. :-(

**Comment by whit537** [comment]
I moved the `decode_raw` refactor to a new PR: #491. Ready for review here, @pjz @Changaco @techtonik et al.

File: aspen/http/resource.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+from .response import Response
+from ..simplates import Simplate, SimplateDefaults, SimplateException
+
+
+class Static(object):
+    """Model a static HTTP resource.
```
Comment (techtonik) [inline aspen/http/resource.py]: What does mean "model a resource"?

Comment (whit537) [inline aspen/http/resource.py]: It means to bring [web resources](https://en.wikipedia.org/wiki/Web_resource) under the Aspen [object model](https://en.wikipedia.org/wiki/Object_model). An object model is:

> A collection of objects or classes through which a program can examine and manipulate some specific parts of its world. In other words, the object-oriented interface to some service or system. Such an interface is said to be the object model of the represented service or system.

Comment (techtonik) [inline aspen/http/resource.py]: Yea, but this is too abstract. Is it an object that represents static resource on disk, in template, or in web request?

**Pull request merged**